tests/common/mod.rs

```diff
@@ -5,6 +5,110 @@ use web_sys::{
     wasm_bindgen::JsValue,
 };
 
+async fn time_it<R>(cb: impl std::future::Future<Output = R>) -> (f64, R) {
+    let now = {
+        use web_sys::{
+            js_sys::{global, Reflect},
+            wasm_bindgen::JsCast,
+            Performance,
+        };
+
+        let performance = Reflect::get(&global(), &"performance".into())
+            .unwrap()
+            .unchecked_into::<Performance>();
+        move || performance.now()
+    };
+    let start_ms = now();
+    let ret = cb.await;
+    let elapsed_ms = now() - start_ms;
+    (elapsed_ms, ret)
+}
+
+#[wasm_bindgen_test]
+async fn close_and_delete_before_reopen() {
+    // tracing_wasm::set_as_global_default();
+    // std::panic::set_hook(Box::new(console_error_panic_hook::hook));
+
+    const DATABASE_NAME: &str = "close_and_delete_before_reopen";
+
+    for _ in 0..10 {
+        let factory = Factory::get().unwrap();
+
+        let (delete_duration_ms, _) = time_it(async {
+            factory.delete_database(DATABASE_NAME).await.unwrap();
+        })
+        .await;
+        // Deleting the database should be almost instantaneous in theory.
+        // However this operation will hang as long as the database is still opened,
+        // which can last for 10s of seconds if our code forget to close is (in
+        // which case the close will only occur when the underlying javascript
+        // object got garbage collected...).
```

Comment:
Oh I was thinking of timing a full loop of 10 open/delete cycles and asserting it's less than eg. 10 seconds, so that on average it'll really take much less but without the implementation it'd take literally hundreds of seconds.

This being said it's not at all a blocker, and I'll land if this is the last remaining thing :smile: